jujugui/static/gui/src/app/components/user-profile/create-model-button/create-model-button.js

```diff
@@ -0,0 +1,158 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('create-model-button', function() {
+
+  juju.components.CreateModelButton = React.createClass({
+    propTypes: {
+      addNotification: React.PropTypes.func.isRequired,
+      className: React.PropTypes.string,
+      env: React.PropTypes.object.isRequired,
+      hideConnectingMask: React.PropTypes.func.isRequired,
+      jem: React.PropTypes.object,
+      showConnectingMask: React.PropTypes.func.isRequired,
+      switchModel: React.PropTypes.func.isRequired,
+      user: React.PropTypes.object
+    },
+
+    getInitialState: function() {
+      return {
+        createNewModelActive: false
+      };
+    },
+
+    getDefaultProps: function() {
+      // Just in case broadcastStatus isn't passed in (e.g., in tests), calls
+      // to it should not fail, so default to an empty function.
+      return {
+        className: ''
+      };
+    },
+
+    /**
+      Fetches the model name from the modelName ref and then creates a model
+      then switches to it.
+      @method createAndSwitch
+      @param {Object} e The event handler from a form submission.
+    */
+    createAndSwitch: function(e) {
+      if (e && e.preventDefault) {
+        e.preventDefault();
+      }
+      // The supplied new model name needs to be valid.
+      var modelName = this.refs.modelName;
+      if (!modelName.validate()) {
+        // Only continue if the validation passes.
+        return;
+      }
+      var props = this.props;
+      // Because this will automatically connect to the model lets show
+      // the connecting mask right now.
+      props.showConnectingMask();
+      // XXX This is only for the JIMM flow.
+      props.env.createModel(
```

Comment:
This is now incorrect once #1946 lands. It now requires the controller api instance to create the model. See https://github.com/juju/juju-gui/pull/1946/files#diff-03aef66566d648c9516ccc2d653dfa24R178